src/dodal/utils.py

```diff
@@ -91,7 +99,7 @@ def wrapper(*args, **kwds) -> T:
 
 def make_all_devices(
     module: Union[str, ModuleType, None] = None, **kwargs
-) -> Dict[str, HasName]:
+) -> dict[str, AnyDevice]:
```

Comment:
```suggestion
) -> Dict[str, AnyDevice]:
```

dict can only be used from python 3.10 onwards, and dodal supports 3.8+

Alternatively: add `from future import __annotations__` at the top of the file